one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -106,23 +108,23 @@ public void onError(String commandName, Exception exception) {
                 template.convertAndSend("/topic/install-status", new StatusMessage(lastPercentageComplete, message));
                 logger.error(message, exception);
                 dataHubConfigurationException[0] = exception;
-                environmentService.setProjectDirectory(originalDirectory);
             }
         };
         // setting the project directory will resolve any relative paths
         try {
             environmentService.setProjectDirectory(payload.get("directory").asText(""));
```

Comment:
String directory = payload.get("directory");
If (StringUtils.isEmpty(directory)) { // input validation should be checked in UI side, and here just in case
    //maybe logger.error(...)
    return payload;
}
hubConfig.createProject(directory);
...
we had better put the line  "environmentService.setProjectDirectory(directory)" to the end when dhf has been succcessfully installed.